### DISCUSSION TITLE
Can we reduce bundle size using Webpack?

### DISCUSSION THREAD

**Discussion by mgulyas42** [haraka/Haraka#3299] [General]
We're exploring options to optimize the bundle size of Haraka. Our goal is to ensure that Haraka operates within a few megabytes.

We have a `node_modules` directory with size of 130 MB.  Therefore we also need to webpack the node_modules, which presents further challenges regarding the plugins. (If we want to add a new plugin, it's likely that we must bundle again, but its OK)

Can you help us how can we bundle Haraka or just reducing its size?

**Comment by mgulyas42** [comment]
I've found that the `haraka-plugin-geoip`  contains tests, which are quite large in size. (69MB)

**Reply by msimerson** [reply]
Yep, it _did_. But look what happened [yesterday](https://www.npmjs.com/package/haraka-plugin-geoip):

<img width="259" alt="Screenshot 2024-04-04 at 8 06 05 AM" src="https://github.com/haraka/Haraka/assets/261635/9ff571a0-bfa9-44fa-8f35-f1c7919c61f1">

**Reply by mgulyas42** [reply]
It's a magic, thank you @msimerson !

**Comment by mgulyas42** [comment]
The original question was about to bundle the haraka, so it should be still open

**Reply by msimerson** [reply]
You're welcome to try but that's not the low hanging fruit for reducing Haraka's size. Not publishing the tests to NPM has already reduced the size by more than any form of compression could.

I'm working my way through the Haraka packages as part of https://github.com/orgs/haraka/projects/2
